File: Python/largest-palindrome-product.py
class Solution:
    def largestPalindrome(self, n):
        """
        :type n: int
        :rtype: int
        """
        upper = pow(10, n) - 1

        for i in range(upper):
            i = upper - i
            if i <= upper / 10:
                break
            palindrome = int(str(i) + str(i)[::-1])

            for j in range(upper):
                j = upper - j
                if j <= upper / 10 or palindrome / j > upper:
                    break
                if palindrome % j == 0:
                    return palindrome % 1337

        return 9

File: Swift/largest-palindrome-product.swift
class Solution {
    func largestPalindrome(_ n: Int) -> Int {
        let upper = Int(pow(10, Double(n)) - 1)

        for i in 0..<upper {
            let i = upper - i
            if i <= upper / 10 { break }
            let palindrome = toPalindrome(i)
            for j in 0..<upper {
                let j = upper - j
                if j <= upper / 10 || palindrome / j > upper { break }
                if palindrome % j == 0 {
                    return palindrome % 1337
                }
            }
        }

        return 9
    }

    private func toPalindrome(_ i: Int) -> Int {
        let iStr = String(i)
        let str = iStr + String(iStr.reversed())
        return Int(str)!
    }
}
